client/dashboard/app/router.tsx

```diff
@@ -107,8 +109,12 @@ const sitePerformanceRoute = createRoute( {
 const siteSettingsRoute = createRoute( {
 	getParentRoute: () => siteRoute,
 	path: 'settings',
-	loader: ( { params: { siteSlug } } ) =>
-		queryClient.ensureQueryData( siteSettingsQuery( siteSlug ) ),
+	loader: async ( { params: { siteSlug } } ) => {
+		await queryClient.ensureQueryData( siteSettingsQuery( siteSlug ) );
+		return {
+			breadcrumbItemLabel: __( 'Settings' ),
```

Comment:
Why does this have to be in the loader? Can't it be added as route meta or something?

Reply:
It needs to call `__()`, so it can't be a static data. Also, I imagine other screens might have dynamic breadcrumb item label such as plugin name in the Plugins dashboard (it's the case now in v1 if you go to https://wordpress.com/plugins

BTW as per docs, `meta` has been replaced with https://tanstack.com/router/latest/docs/framework/react/guide/static-route-data.

Reply:
This is my conclusion after spending a day reading the docs and consulting with AI 🥲 but I'm open if you have better ideas! This is literally my first project with TanStack stack.

Reply:
`__()` can be called anywhere, it doesn't have to be called at render time.

But if you really want, you could create a function that maps a route to a label:

```js
function getRouteBreadcrumbLabel( route ) {
    switch ( route.path ) {
        case 'setting': return __( 'Settings' );
    }
}
```

The loader function doesn't seem meant for this

Reply:
How would that work with dynamic label such as plugins (as I mentioned above)

<img width="690" alt="image" src="https://github.com/user-attachments/assets/4f526e62-3344-4232-9745-79d9ef66578c" />

We will need to fetch some data first (e.g. in the loader) before being able to return the label.

Reply:
Can you point me to the designs for this?

Reply:
The design is not available yet in v2 actually. I just wanted to prepare for this case in case we want to bring this pattern to v2 😬

Reply:
I think yeah we can try to put that in `staticData` (formerly `meta`) first and try to figure out the dynamic case when it's actually needed in the design...

> __() can be called anywhere, it doesn't have to be called at render time.

I was afraid the translation might not be ready yet when we construct the routing, but I'm happy to be wrong. I can try.

Reply:
Yeah, imo we shouldn't think about use cases that might not exist. If for now a simple link with a slash works, then imo that's good for now

Reply:
> I was afraid the translation might not be ready yet when we construct the routing, but I'm happy to be wrong. I can try.

For `@wordpress/i18n` in GB, I don't think it matters, I'm not sure about Calypso. But if you want to be sure it could be a function that maps it instead that is called during render

Reply:
> Why does this have to be in the loader? Can't it be added as route meta or something?

Thanks for the feedback! I ended up putting the label inside `staticData` instead, with a small caveat:

> `__()` can be called anywhere, it doesn't have to be called at render time.

It doesn't work with the approach in:

- https://github.com/Automattic/wp-calypso/pull/103299

as the translation is not ready yet when the routing is constructed, so I made it a function instead, e.g.:

```js
const siteSettingsRoute = createRoute( {
	...
	staticData: {
		breadcrumbItemLabel: () => __( 'Settings' ),
	},
```